packages/automerge-repo-storage-nodefs/src/index.ts

```diff
@@ -82,59 +64,42 @@ export class NodeFSStorageAdapter extends StorageAdapter {
        and could probably be simplified. */
 
     const dirPath = this.getFilePath(keyPrefix)
-    const cacheKeyPrefixString = cacheKey(keyPrefix)
-
-    // Get the list of all cached keys that match the prefix
-    const cachedKeys: string[] = Object.keys(this.cache).filter(key =>
-      key.startsWith(cacheKeyPrefixString)
-    )
+    const cachedKeys = this.cachedKeys(keyPrefix)
 
     // Read filenames from disk
-    let diskFiles
-    try {
-      diskFiles = await walkdir(dirPath)
-    } catch (error) {
-      if (error.code === "ENOENT") {
-        // Directory not found, initialize as empty
-        diskFiles = []
-      } else {
-        throw error
-      }
-    }
+    const diskFiles = await walkdir(dirPath)
 
     // The "keys" in the cache don't include the baseDirectory.
     // We want to de-dupe with the cached keys so we'll use getKey to normalize them.
-    const diskKeys: string[] = diskFiles
-      .flat(Infinity) // the walk function returns a nested array
-      .map(fileName =>
-        this.getKey([path.relative(this.baseDirectory, fileName)])
-      )
+    const diskKeys: string[] = diskFiles.map((fileName: string) =>
+      getKey([path.relative(this.baseDirectory, fileName)])
+    )
 
     // Combine and deduplicate the lists of keys
     const allKeys = [...new Set([...cachedKeys, ...diskKeys])]
 
     // Load all files
-    return Promise.all(
+    const result = await Promise.all(
       allKeys.map(async keyString => {
         const key: StorageKey = keyString.split(path.sep)
-        return {
-          data: await this.load(key),
-          key,
-        }
+        const data = await this.load(key)
+        return { data, key }
       })
     )
+
+    return result
   }
 
   async removeRange(keyPrefix: string[]): Promise<void> {
```

Comment:
we should refuse to run this without a key value; pretty sure someone accidentally passing in undefined / empty array would delete everything they have

Reply:
ok - that should be a separate PR